File: csharp/Game/Enemy.cs
namespace Game {

public class Enemy(string name, int health) : Character(name, health) {

        // Similar to the Player class, the 'override' keyword is used in C# to override the base class method.
        // This is equivalent to the @Override annotation in Java.
        public override void Attack() {
        Console.WriteLine($"{Name} attacks with a claw!");
    }
}

}

File: java/game/src/main/java/example/Enemy.java
package example;

// This class showcases method overriding in Java.
// It extends the Character class and provides a specific implementation for the attack method.
public class Enemy extends Character {
    public Enemy(String name, int health) {
        super(name, health);
    }

    @Override
    public void attack() {
        System.out.println(name + " attacks with claws!");
    }
}

